src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -1,22 +1,34 @@
 package core.basesyntax.dao.machine;
 
 import core.basesyntax.dao.AbstractDao;
+import core.basesyntax.exceptions.DataProcessingException;
 import core.basesyntax.model.machine.Machine;
+import java.time.LocalDate;
 import java.util.List;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
 
-public class MachineDaoImpl extends AbstractDao implements MachineDao {
+public class MachineDaoImpl extends AbstractDao<Machine> implements MachineDao {
     protected MachineDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Machine save(Machine machine) {
-        return null;
+        return super.save(machine);
     }
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
-        return null;
+        int year = LocalDate.now().getYear() - age;
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Machine "
+                    + "WHERE year > :year", Machine.class)
```

Comment:
Older than, means that production year is lower than year in parameter. In your case year is greater.
Try make test with your solution